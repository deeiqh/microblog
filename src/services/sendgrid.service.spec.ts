import { faker } from "@faker-js/faker";
import { SendgridService } from "./sendgrid.service";

jest.mock("@sendgrid/mail", () => {
  return {
    setApiKey: jest.fn(),
    send: jest.fn().mockImplementation(() =>
      Promise.resolve([
        {
          statusCode: 200,
          headers: {},
          body: "",
        },
      ])
    ),
  };
});

describe("SendgridService", () => {
  describe("sendEmail", () => {
    it("should send an email", async () => {
      const to = faker.internet.email();
      const text = faker.lorem.words();
      const subject = faker.lorem.word();

      const result = await SendgridService.sendEmail({
        to,
        html: text,
        text,
        subject,
      });

      expect(result).toMatchSnapshot();
    });
  });
});
