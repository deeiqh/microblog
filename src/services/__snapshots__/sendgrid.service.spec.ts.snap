// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendgridService sendEmail should send an email 1`] = `
[
  {
    "body": "",
    "headers": {},
    "statusCode": 200,
  },
]
`;
